mustache.go

```diff
@@ -31,19 +31,31 @@ func (es ErrorSlice) Error() string {
 	return b.String()
 }
 
+func tryAddNode(visitedNodes map[node]struct{}, newNode node) error {
+	if _, ok := visitedNodes[newNode]; !ok {
+		visitedNodes[newNode] = struct{}{}
+		return nil
+	}
+	return fmt.Errorf("the node %s has been already visited in the mustache node graph", newNode.String())
```

Comment:
maybe use `%q` ? And maybe an error message that is more user friendly. Perhaps:

`A circular lookup reference has been detected at value %q`

Reply:
yep will do that :)